src/controllers/volunteer.controller.ts

```diff
@@ -46,3 +47,57 @@ export const getVolunteers = async (req: Request, res: Response) => {
     return res.status(500).json({ error: error.message });
   }
 };
+
+export const approveVolunteer = async (req: Request, res: Response) => {
+  const { volunteerId } = req.params;
+
+  if (!volunteerId) {
+    return res.status(400).json({ error: 'No volunteer id provided.' });
+  }
+
+  if (!mongoose.Types.ObjectId.isValid(volunteerId)) {
+    return res.status(400).json({ error: 'Invalid volunteer ID.' });
+  }
+
+  try {
+    const volunteerObj = await Volunteer.findById(volunteerId);
+
+    if (!volunteerObj) {
+      return res.status(400).json({ error: 'cannot find volunteer object' });
+    }
+
```

Comment:
Should make this endpoint multi use? Maybe turn it into `export const changeVolunteerApprovalStatus = ...` and allow the user to send up a body parameter containing the new status. Therefore the endpoint can be used for approving and denying instead of having to write two different endpoints.